Cannot install Coq 8.14 via opam on macOS due to sandboxing + dune 2.x caching

<!-- Thank you for reporting a bug to Coq! -->

#### Description of the problem

Running `opam install coq.8.14.0` with a fresh opam install on macOS 11.6 fails for me:

```
[ERROR] The compilation of coq.8.14.0 failed at "./configure -configdir /Users/tchajed/.opam/default/lib/coq/config
        -prefix /Users/tchajed/.opam/default -mandir /Users/tchajed/.opam/default/man -docdir
        /Users/tchajed/.opam/default/doc/coq -libdir /Users/tchajed/.opam/default/lib/coq -datadir
        /Users/tchajed/.opam/default/share/coq -coqide no -native-compiler no".

#=== ERROR while compiling coq.8.14.0 =========================================#
# context     2.1.0 | macos/x86_64 | ocaml.4.12.0 | https://opam.ocaml.org#c7b1bb0d
# path        ~/.opam/default/.opam-switch/build/coq.8.14.0
# command     ~/.opam/opam-init/hooks/sandbox.sh build ./configure -configdir /Users/tchajed/.opam/default/lib/coq/config -prefix /Users/tchajed/.opam/default -mandir /Users/tchajed/.opam/default/man -docdir /Users/tchajed/.opam/default/doc/coq -libdir /Users/tchajed/.opam/default/lib/coq -datadir /Users/tchajed/.opam/default/share/coq -coqide no -native-compiler no
# exit-code   1
# env-file    ~/.opam/log/coq-31916-bb6357.env
# output-file ~/.opam/log/coq-31916-bb6357.out
### output ###
# Error: too many retries (100): waiting for beacon file
# "/Users/tchajed/.cache/dune/db/runtime/dune-cache-daemon/port" to be locked
# Error: empty output starting cache
```

If I start the dune cache manually with `dune cache start`, I get a slightly different error message:

```
# Error: connect: : Operation not permitted
```

`opam install coq.8.13.2` still works, and I can compile Coq from source, and Homebrew can still build the Coq bottle, so the issue seems to be a regression in Coq's opam build process specifically (maybe due to sandboxing?).

#### Coq Version

- opam 2.1.0
- dune 2.9.1
- coq 8.14.0
- macOS 11.6

--- a/INSTALL.md
+++ b/INSTALL.md
@@ -39,6 +39,9 @@ To compile Coq yourself, you need:
 
 - [optional] GNU Make (version >= 3.81)
 
+See [below](#Known-Problems) for a discussion of platform-specific
+issues with dependencies.
+
 Primitive floating-point numbers require IEEE-754 compliance
 (`Require Import Floats`). Common sources of incompatibility
 are checked at configure time, preventing compilation. In the
@@ -116,3 +119,11 @@ like `COQLIB="some path"`, that is a variable name followed by `=` and
 a string that follows OCaml's escaping conventions. This feature can be
 used by installers of binary package to make Coq aware of its installation
 path.
+
+Known problems
+--------------
+
+- On OSX, when using Opam with sandboxing and Dune < 3.0 with caching
+  enabled, Coq may fail to install, due to Opam's sandboxing
+  restricting the permissions needed for the Dune 2.x caching
+  daemon. See https://github.com/coq/coq/issues/15138 more details.

